apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/internal/json/JsonScope.kt

```diff
@@ -46,14 +46,18 @@ object JsonScope {
    * pathIndices when doing so isn't useful.
    */
   fun getPath(stackSize: Int, stack: IntArray, pathNames: Array<String?>, pathIndices: IntArray): String {
-    val result = StringBuilder().append('$')
+    val result = StringBuilder()
+    var isRoot = true
     for (i in 0 until stackSize) {
       when (stack[i]) {
-        EMPTY_ARRAY, NONEMPTY_ARRAY -> result.append('[').append(pathIndices[i]).append(']')
+        EMPTY_ARRAY, NONEMPTY_ARRAY -> result.append('.').append(pathIndices[i])
```

Comment:
is this right?

Reply:
Not really but this is what FileUpload expects... I put a comment there https://github.com/apollographql/apollo-android/blob/dfc11b8dc51fe12a37e328711f789af3d705c36d/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/internal/json/BufferedSinkJsonWriter.kt#L71

Reply:
Actually, I'm not sure if one is better than the other now that I think about it. 

Nothing prevents some object to have a `"[0]"` key....